### DISCUSSION TITLE
Paid API access with Bitcoin's Lightning Network (and LNbits)

### DISCUSSION THREAD

**Discussion by talvasconcelos** [oobabooga/text-generation-webui#1811] [Ideas]
I'm a colaborator at [LNbits](https://github.com/lnbits/lnbits), and part of my work is creating extensions. I'd like to make an LNbits extension where `User A` has oobabooga running with `--public-api` and adds the URL in the extension. The extension gives another URL, which will be a proxy, the other users can use after pay. Let's say `User B` would like to use AI and these cool things but, he doesn't have a powerfull GPU/CPU or even a computer. He adds some funds upfront into the LNbits extension `User A` has set up, and uses that API URL on something like TavernAI, and gets charged X sats per API call. 

`User A` gets some bitcoin for his service, `User B` gets to chat with some nice characters on TavernAI.

This could then become a market competition, different models, filtered vs unfiltered, large vs small models, etc... There could also be an extension for characters!

I'm mentioning TavernAI firstly because it's what I've seen and because I couldn't "share" webui url! I'm open to comments and critics, idea validation, etc... Not even sure if such it's possible with oobabooga!

**Comment by eucdee** [comment]
gradio has a access control system, i didnt test it yet but i guess you could build on that.
there are also telegram bots in the wild that use the gpt4 api and prepaid charge for calls like what you want to archive...
probably best to use monthly subscribtion like everybody else..
LNbits looks good and comes with account system, couldnt hurt if ooba gets a plugin to support it...
not sure if i would limit it to lightning btc only...

these days the rules are simple, if you can dream it, you can build it :)
